tests/internals/initial_delay_cooldownperiod/initial_delay_cooldownperiod_test.go

```diff
@@ -0,0 +1,128 @@
+//go:build e2e
+// +build e2e
+
+package initial_delay_cooldownperiod_test
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+const (
+	testName = "initial-delay-cooldownperiod-test"
+)
+
+var (
+	testNamespace    = fmt.Sprintf("%s-ns", testName)
+	deploymentName   = fmt.Sprintf("%s-deployment", testName)
+	scaledObjectName = fmt.Sprintf("%s-so", testName)
+
+	now   = time.Now().Local()
+	start = (now.Minute() + 10) % 60
+	end   = (start + 1) % 60
+)
+
+type templateData struct {
+	TestNamespace    string
+	DeploymentName   string
+	ScaledObjectName string
+	StartMin         string
+	EndMin           string
+}
+
+const (
+	deploymentTemplate = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    deploy: {{.DeploymentName}}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      pod: {{.DeploymentName}}
+  template:
+    metadata:
+      labels:
+        pod: {{.DeploymentName}}
+    spec:
+      containers:
+        - name: nginx
+          image: 'nginxinc/nginx-unprivileged'
+`
+
+	scaledObjectTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  cooldownPeriod: 5
+  minReplicaCount: 0
+  initialDelayCooldownPeriod: 120
+  advanced:
+    horizontalPodAutoscalerConfig:
+      behavior:
+        scaleDown:
+          stabilizationWindowSeconds: 15
+  triggers:
+  - type: cron
+    metadata:
+      timezone: Etc/UTC
+      start: {{.StartMin}} * * * *
+      end: {{.EndMin}} * * * *
+      desiredReplicas: '10'
+`
+)
+
+func TestScaler(t *testing.T) {
+	// setup
+	t.Log("--- setting up ---")
+
+	// Create kubernetes resources
+	kc := GetKubernetesClient(t)
+	data, templates := getTemplateData()
+
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 1, 60, 1),
+		"replica count should be 1 after 1 minute")
+
+	// test scaling
+	testScaleIn(t, kc)
+
+	// cleanup
+	DeleteKubernetesResources(t, testNamespace, data, templates)
+}
```

Comment:
I think that we could simplify this test just deploying a Deployment with 2 instances and a ScaledObject with a cron scaler that always return 0. Then we just need to assert that the deployment doesn't change during some time and then is scales to zero. something like this assuming 2 instances for the workload(using correct code, I'm writing it from memory)
```
CreateKubernetesResources(t, kc, testNamespace, data, templates) // Deploy the workload + ScaledObject
AssertReplicaCountNotChangeDuringTimePeriod(t, kc, deploymentName, testNamespace, 2, 90) // Ensure that the workload has 2 instances during 90 seconds
assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 0, 6, 10),
		"replica count should be %d after 1 minute", 0) // Assert that the workload is scaled to zero after the initial cooldown period
```